File: files/2025-11-30/CSEC 30112025.py
import pandas as pd
import re

path = "2025-11-30/Challenge 81.xlsx"
input = pd.read_excel(path, usecols="B", nrows = 4, skiprows = 2)
test = pd.read_excel(path, usecols="D", nrows=4, skiprows=2).rename(columns=lambda c: c.replace('.1', ''))

def clean_text(s):
    return s if pd.isnull(s) else " ".join(re.sub(r"<.*?>", "", str(s)).replace("&nbsp;", "").split())

result = input.map(clean_text)

print(result.equals(test)) # True

File: files/2025-11-30/CSEC 30112025.R
library(tidyverse)
library(readxl)

path <- "2025-11-30/Challenge 81.xlsx"
input <- read_excel(path, range = "B2:B6")
test <- read_excel(path, range = "D2:D6")

result = input %>%
  mutate(across(everything(), ~ str_remove_all(., "<.*?>"))) %>%
  mutate(across(everything(), ~ str_replace_all(., "&nbsp;", ""))) %>%
  mutate(across(everything(), ~ str_squish(.)))

all.equal(result$Problem, test$Solution)
# [1] TRUE
